File: CSharp part2/01. Arrays/04.MaximalSequence/MaximalSequence.cs
// Write a program that finds the maximal sequence of equal elements in an array.

using System;
using System.Linq;
using System.Collections.Generic;

class MaximalSequence
{
    static void Main()
    {
        Console.Write("Enter array elements(separated by a space):");
        string number = Console.ReadLine();
        string[] splitNumbers = number.Split(' ');
        int[] arr = new int[splitNumbers.Length];
        for(int i=0;i<arr.Length;i++)
        {
            arr[i] = int.Parse(splitNumbers[i]);
        }
        int currIndex = 0;
        int currLength = 0;
        int bestLength = 0;
        int bestElement = 0;
        for (int i = 0; i < arr.Length; i++)
        {
            if (arr[i] == arr[currIndex])
            {
                currLength++;
            }
            else
            {
                currIndex = i;
                currLength = 1;
            }
            if (currLength > bestLength)
            {
                bestLength = currLength;
                bestElement = arr[i];
            }
        }
        for (int i = 0; i < bestLength; i++)
        {
            Console.Write("{0},", bestElement);
        }
        Console.WriteLine();
    }
}

File: JavaScript-Fundamentals/Arrays/03.MaximalSequence/MaximalSequence.js
// Write a script that finds the maximal sequence of equal elements in an array.

var jsConsole;
var btn = document.getElementById('findMaxSequence');
btn.onclick = findMaximalSequence;

function findMaximalSequence () {
	var array = jsConsole.read('#array').split(" ");

	var currIndex = 0;
	var currLength = 0;
	var bestLength = 0;
	var bestElement = 0;

	for (var i = 0; i < array.length; i++) {
		if (array[i] == array[currIndex]) {
			currLength++;
		} else {
			currIndex = i;
			currLength = 1;
		}
		if (currLength > bestLength) {
			bestLength = currLength;
			bestElement = array[i];
		}
	}

	for (var j = 0; j < bestLength; j++){
		jsConsole.write(bestElement + " ");
		console.log(bestElement + " ");
	}

}